evennia/web/api/permissions.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework import permissions
+
+from django.conf import settings
+
+
+class EvenniaPermission(permissions.BasePermission):
+    """
+    A Django Rest Framework permission class that allows us to use
+    Evennia's permission structure. Based on the action in a given
+    view, we'll check a corresponding Evennia access/lock check.
+    """
+    # subclass this to change these permissions
+    MINIMUM_LIST_PERMISSION = settings.REST_FRAMEWORK["DEFAULT_LIST_PERMISSION"]
+    MINIMUM_CREATE_PERMISSION = settings.REST_FRAMEWORK["DEFAULT_CREATE_PERMISSION"]
+    view_locks = settings.REST_FRAMEWORK["DEFAULT_VIEW_LOCKS"]
+    destroy_locks = settings.REST_FRAMEWORK["DEFAULT_DESTROY_LOCKS"]
+    update_locks = settings.REST_FRAMEWORK["DEFAULT_UPDATE_LOCKS"]
+
+    def has_permission(self, request, view):
+        """
+        This method is a check that always happens first. If there's an object involved,
+        such as with retrieve, update, or delete, then the has_object_permission method
+        is also called if this returns True. If we return False, a permission denied
+        error is raised.
```

Comment:
Nice docstring, but it is not on the standard form for Evennia docstrings. Here is an example of rewriting it on the Evennia form: 

```
Determine if the permission check is passed for a given view. 

Args: 
    request (Request): The incoming request object. 
    view (View): The django view we are checking permission for. 

Returns:
    bool: If permission is granted or not. If we return False here, a PermissionDenied error will be raised
         from the view. 

Notes:  
        This method is a check that always happens first. If there's an object involved,
        such as with retrieve, update, or delete, then the has_object_permission method
        is called after this, assuming this returns `True`. 
```